flow-server/src/main/resources/plugins/theme-loader/theme-loader.js

```diff
@@ -3,7 +3,8 @@ const fs = require('fs');
 const path = require('path');
 
 // Collect groups [url(] [ |'|"]optional './|../', file part and end of url
-const urlMatcher = /(url\()(\'|\")?(\.\/|\.\.\/)(\S*)(\2\))/g;
+// takes into account that whitespace in file name and query parameters are valid
+const urlMatcher = /(url\()(['"])?(\.\/|\.\.\/)([^?'")]+)((\?\S*)?\2\))/g;
```

Comment:
```suggestion
const urlMatcher = /(url\()(['"])?(\.\/|\.\.\/)([^?\2)]+)(\?\S*)?(\2\))/g;
```
Would do a minor update for the regex so that we get end line and query parameter in their own groups (add the query to line 35) also change the internal `^?'"` to `^?\2`to enable the unlikely strings mixing the characters like: `url('./fonts/Roboto/roboto-v29-latin ext_"latin"-regular.woff2')`

Reply:
>  also change the internal `^?'"` to `^?\2`

I had it like this, but changed it since I noticed that IDEA regex validation would not understand it and https://regexr.com/69jqg said that `\2` inside the `[^]` would "matches a STX character (char code 2)". 
It didn't occur to me how to use the backreference inside `[^]` so I just changed it to this but it is problematic indeed.
I'll need to investigate this more and think about what to do with `"` and `'` inside the url.

> Would do a minor update for the regex so that we get end line and query parameter in their own groups (add the query to line 35)

_end line_ ? Not sure I'm sure what you mean with that - the string can even continue on the same line with e.g. `format('woff2')` so please clarify what you mean. Also what is the point of separating the group, as the optional query part and the `")` ending need to be added back anyway.

Reply:
Now the end result is a weird 
```
Group 5 |  ?v=1.12')
Group 6 |  ?v=1.12
```
so if you can get it to not generate 6 groups then it's fine to have the full end result, but the reading is hard with the current version.

I didn't note that it says matches the explicit character 2 instead of the group. I guess we can forget that part from this at least as I don't know if it is an actual use case.

Reply:
Testing this, it seems to work but I'm not sure I understand _why_ it works 😅 as it claims that it is not matching
![image](https://user-images.githubusercontent.com/1478381/141946919-05a16b2b-f4ca-486b-9e54-b5ffc7023355.png)

Reply:
regex101 tester also shows that it works, but says that it matches a character 2 I guess the problem is if you have `2)` in the string? as ") doesn't match a url starting with ' and vice versa.

Reply:
Well if I change it from `.woff2` to `.woff` then it still works ...

Reply:
The current one just has the problem of `url('./fonts/Roboto/roboto-v29-latin-ext_la2)tin-regular.woff?v=1.12')` not matching

Reply:
I guess the best would be to just get the whole url (query and all) and then for the file check drop any querypart, so have 2 regexes as this case seems too complex to make with a one liner (or it gets too complex to maintain)

Reply:
At this point, it feels more painful to try to split it into two "reasonable" regexp instead of just going with one that works even when it is very complex: https://regexr.com/69jqg

Reply:
Why would we create our own URL parser when there are many perfectly fine ones out there?

Reply:
Where is a new URL parser being created ? This PR tries to fix a broken regex that picks any `url( ... )` strings from css files and then replaces a part of the url address.

If there is a better way to do the url replacement, please suggest the way to do it.

Reply:
In this case I would rather not support all the random cases as now the latest regex takes
![image](https://user-images.githubusercontent.com/2726090/142137756-046f5742-da6e-4ce5-ac6e-8a934737b2c7.png)
in comparison to the simpler that fails in special cases where 2) is in the filename as the difference in complexity is quite noticable
![image](https://user-images.githubusercontent.com/2726090/142138136-20e4e05d-51ea-4eb4-b3cd-1c928300f70e.png)

Reply:
Ok, good point. The JS code could then undergo further refactoring and for sure it would make sense to use something that is already available.

> I would rather not support all the random cases

Here I kind of disagree - whatever the solution is, it should support valid url() function parameters. In this case though, it can be limited to the fact that it should also be a valid file name as it only touches relative file urls.